### ISSUE TITLE
May want to use rmdir/remove/nftw instead of system() for remove directory

### ISSUE THREAD

**Issue by stevetranby** [cocos2d/cocos2d-x#16817]
Either use unlink while recursively readdir. Or I think the nftw code from the Apple-specific file utils may work for Linux as well?

While Stackoverflow is not the best to use for final code, here's topics for reference.

http://stackoverflow.com/questions/5467725/how-to-delete-a-directory-and-its-contents-in-posix-c
https://stackoverflow.com/questions/3184445/how-to-clear-directory-contents-in-c-on-linux-basically-i-want-to-do-rm-rf
http://stackoverflow.com/questions/2256945/removing-a-non-empty-directory-programmatically-in-c-or-c

**Comment by minggo** [comment]
as the link says, unlink is used for non-directories, so i think we can use nftw codes from the Apple-specific codes.

**Comment by stevetranby** [comment]
Ah yeah, remove/rmdir or if all platforms support it ntfw.

Reference to create PR:

system()
https://github.com/cocos2d/cocos2d-x/blob/4b3525f7a828bceb382f6771be333ce83b1cb601/cocos/platform/CCFileUtils.cpp#L1155

nftw
https://github.com/cocos2d/cocos2d-x/blob/c0e1e913739404ad4213d3af8c3a1ea468c06cbb/cocos/platform/apple/CCFileUtils-apple.mm#L288

**Comment by minggo** [comment]
I think only unix similar systems supports nftw

**Comment by stevetranby** [comment]
The windows code seems to call the one that executes the system('rm -rf') code. I wasn't offering solution, just reference notes in case someone wants to try and consolidate. May be another one of those "even is system() is no longer a recommended method going forward, just use it only for all v3.x versions"

Anyway, you could always just make sure v4.x is based on a minimum build requirement of c++17 filesystem TS, heh.

**Comment by minggo** [comment]
I think we can use nftw for all unix like systems and remove the Apple-specific implementation. 

We haven't use c++14, when will c++17 be popular :)

**Comment by stevetranby** [comment]
Yeah, that's why I recommended it, haha. Make v4 forward thinking, heh.
